lib/libc/Kconfig

```diff
@@ -15,6 +15,14 @@ config NEWLIB_LIBC
 	  Build with newlib library. The newlib library is expected to be
 	  part of the SDK in this case.
 
+config NEWLIB_LIBC_NANO
+	bool "Build with newlib-nano c library"
+	select NEWLIB_LIBC
+	help
+	  Build with newlib-nano library, for small embedded apps.
+	  The newlib-nano library for ARM embedded processors is a part of the
+	  GNU Tools for ARM Embedded Processors.
```

Comment:
As the build scripts are organized, nano is a configuration/variant of newlib.

Generally we should use 'depends on' and not 'select' whenever possible,
so let us change this from 'select NEWLIB_LIBC' to 'depends on 'NEWLIB_LIBC'.

Preferably, we would write this 'depends on' as an if statement that would cover nano, and the below three options as well.